File: 2094-Finding-3-Digit-Even-Numbers/2094.swift
class Solution {
    func findEvenNumbers(_ digits: [Int]) -> [Int] {

    	var freq = [Int](repeating: 0, count: 10)
    	var requiredDigits = [Int](repeating: 0, count: 10)
    	for d in digits {
        	freq[d] += 1
    	}
    
    	var result = [Int]()
    
    	for i in stride(from: 100, through: 998, by: 2) {
        	let centum = i / 100
        	let decem = (i / 10) % 10
        	let unitas = i % 10
        
        	for j in 0..<10 {
            	requiredDigits[j] = 0
        	}
        	requiredDigits[centum] += 1
        	requiredDigits[decem] += 1
        	requiredDigits[unitas] += 1
        
        	var canFormNumber = true
        	for j in 0..<10 {
            	if requiredDigits[j] > freq[j] {
                	canFormNumber = false 
                	break
            	}
        	}
        
        	if canFormNumber {
            		result.append(i)
        	}
    	}
    
    	return result
    }

}

File: 2094-Finding-3-Digit-Even-Numbers/2094.c
int* findEvenNumbers(int* digits, int digitsSize, int* returnSize) {
    int freq[10] = {0};
    //Number of 3-digit numbers: 999
    //Excluding numbers with leading 0s:(-99) 900
    //Only even numbers: (/2) 450
    int* resultArray = (int*)malloc(sizeof(int) * 450);
    int count = 0;
    bool canFormNumber = true;
    int centum, decem, unitas;

    for (int i = 0; i < digitsSize; i++) {
        freq[digits[i]]++;
    }

    for (int i = 100; i <= 999; i += 2) {
	
        centum = i / 100;
        decem = (i / 10) % 10;
        unitas = i % 10;

        int requiredDigits[10] = {0};
        requiredDigits[centum]++;
        requiredDigits[decem]++; 
        requiredDigits[unitas]++;

        canFormNumber = true;
        for (int j = 0; j < 10; j++) { 
            if (requiredDigits[j] > freq[j]) {
                canFormNumber = false;
                break;
            }
        }
        if (canFormNumber) { 
            resultArray[count++] = i;
        }
    }
    *returnSize = count;
    return resultArray;
}


